MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java

```diff
@@ -779,251 +787,214 @@ public void outputLog(String tag, String msg) {
-            break;
-        }
-        if (trackFormat != null) {
-          loadData.setRequestLabeledBitrate(trackFormat.bitrate);
-        }
+    public BandwidthMetricData onLoadCompleted(String segmentUrl,
+        long bytesLoaded,
+        Format trackFormat) {
+      BandwidthMetricData loadData = super.onLoadCompleted(segmentUrl, bytesLoaded, trackFormat);
+      if (trackFormat != null) {
+        loadData.setRequestLabeledBitrate(trackFormat.bitrate);
       }
       return loadData;
     }
   }
 
-  class BandwidthMetricDash extends BandwidthMetric {
-
-    @Override
-    public BandwidthMetricData onLoadStarted(DataSpec dataSpec, int dataType,
-        Format trackFormat, long mediaStartTimeMs, long mediaEndTimeMs, long elapsedRealtimeMs) {
-      BandwidthMetricData loadData = super.onLoadStarted(dataSpec, dataType, trackFormat,
-          mediaStartTimeMs, mediaEndTimeMs, elapsedRealtimeMs);
-      if (loadData != null) {
-        switch (dataType) {
-          case C.DATA_TYPE_MEDIA:
-            loadData.setRequestEventType("initFragmentLoaded");
-            break;
-          default:
-            break;
-        }
-      }
-      return loadData;
-    }
-
-    @Override
-    public BandwidthMetricData onLoadCompleted(DataSpec dataSpec, int dataType,
-        Format trackFormat, long mediaStartTimeMs, long mediaEndTimeMs, long elapsedRealtimeMs,
-        long loadDurationMs, long bytesLoaded) {
-      BandwidthMetricData loadData = super.onLoadCompleted(dataSpec, dataType, trackFormat,
-          mediaStartTimeMs, mediaEndTimeMs, elapsedRealtimeMs, loadDurationMs, bytesLoaded);
-      if (loadData != null) {
-        switch (dataType) {
-          case C.DATA_TYPE_MANIFEST:
-            loadData.setRequestEventType("manifestLoaded");
-            break;
-          case C.DATA_TYPE_MEDIA:
-            loadData.setRequestEventType("mediaFragmentLoaded");
-            break;
-          default:
-            break;
-        }
-      }
-      return loadData;
-    }
-  }
 
   class BandwidthMetricDispatcher {
 
     private final BandwidthMetric bandwidthMetricHls = new BandwidthMetricHls();
-    private final BandwidthMetric bandwidthMetricDash = new BandwidthMetricDash();
+    //    private final BandwidthMetric bandwidthMetricDash = new BandwidthMetricDash();
+    ArrayList<String> allowedHeaders = new ArrayList<>();
 
-    public BandwidthMetric currentBandwidthMetric() {
-      switch (streamType) {
-        case C.TYPE_HLS:
-          return bandwidthMetricHls;
-        case C.TYPE_DASH:
-          return bandwidthMetricDash;
-        default:
-          break;
-      }
-      return null;
+    public BandwidthMetricDispatcher() {
+      allowedHeaders.add("x-cdn");
+      allowedHeaders.add("content-type");
     }
 
-    public void onLoadError(DataSpec dataSpec, int dataType, IOException e) {
+    public BandwidthMetric currentBandwidthMetric() {
+      // TODO see if we need different data for HLS and for DASH
```

Comment:
What's the conclusion here?

Reply:
For now we have same logic for both dash and HLS, if in the future we need to implement different data for dash and for HLS we would need this code.

Reply:
In the meantime, I would prefer to remove the commented-out code.

Reply:
removed